### ISSUE TITLE
rails should not be a dependency

### ISSUE THREAD

**Issue by quinn** [JSONAPI-Resources/jsonapi-resources#811]
instead it should only depend on the things it actually needs. For example, I doubt sprockets is necessary to make an API work...

**Comment by evolve2k** [comment]
quinn thanks for your comments, agnostic tools definitley can be more useful.

At the moment there is a crazy number of open items on this issue list and ripping out Rails could be a very big task and one that won't just live under a quick issue comment like this.

I think since the repo is explicit that it is "A resource-focused Rails library for developing JSON API compliant servers." that Rails isn't going any time soon.

Would suggest best approach is to either make your own fork and encourage particiapants or possibly raise these concerns in a more comprehensive say blog post and discuss over twitter with key maintainers.

Note to maintainers: Suggest close as not easily actionable (Discussion can be taken to blog posts and open discussion on twitter)

**Comment by quinn** [comment]
This is actually an easy fix. All I had to do was modify the gemspec to
only include the parts of rails that jsonapi resources actually uses.
Everything else worked fine.. Does that make sense? This doesn't require
any refactoring or anything like that.

**Comment by quinn** [comment]
The "rails" gem is just a gemspec that pulls in about a dozen other gems. I don't think any other gem should depend on it, rather it should depend on only the gems it needs...

**Comment by quinn** [comment]
This change is like 3 lines of code. Would a PR be helpful?

**Comment by evolve2k** [comment]
@lgebhardt could u have a look at this one? thx

**Comment by lgebhardt** [comment]
> This change is like 3 lines of code. Would a PR be helpful?

@quinn That would be very much appreciated.

**Comment by quinn** [comment]
@lgebhardt cool, doing now.

**Comment by quinn** [comment]
first pass at this:

https://github.com/cerebris/jsonapi-resources/pull/860

The test suite all passed for me locally. I unfortunately had to vendor minitest-spec-rails because that project has the same problem :( 

When I did this for my project, all I did was remove the rails line from the gemspec. I tried to be a little bit more nuanced here so the test suite could actually run without requiring sprockets, etc.
